mm2src/coins/lightning/ln_platform.rs

```diff
@@ -0,0 +1,583 @@
+        }
+        Timer::sleep(CHECK_FOR_NEW_BEST_BLOCK_INTERVAL).await;
+    }
+}
+
+struct ConfirmedTransactionInfo {
+    txid: Txid,
+    header: BlockHeader,
+    index: usize,
+    transaction: Transaction,
+    height: u32,
+}
+
+impl ConfirmedTransactionInfo {
+    fn new(txid: Txid, header: BlockHeader, index: usize, transaction: Transaction, height: u32) -> Self {
+        ConfirmedTransactionInfo {
+            txid,
+            header,
+            index,
+            transaction,
+            height,
+        }
+    }
+}
+
+pub struct Platform {
+    pub coin: UtxoStandardCoin,
+    /// Main/testnet/signet/regtest Needed for lightning node to know which network to connect to
+    pub network: BlockchainNetwork,
+    /// The best block height.
+    pub best_block_height: AtomicU64,
+    /// Default fees to and confirmation targets to be used for FeeEstimator. Default fees are used when the call for
+    /// estimate_fee_sat fails.
+    pub default_fees_and_confirmations: PlatformCoinConfirmations,
+    /// This cache stores the transactions that the LN node has interest in.
+    pub registered_txs: PaMutex<HashMap<Txid, HashSet<Script>>>,
+    /// This cache stores the outputs that the LN node has interest in.
+    pub registered_outputs: PaMutex<Vec<WatchedOutput>>,
+    /// This cache stores transactions to be broadcasted once the other node accepts the channel
+    pub unsigned_funding_txs: PaMutex<HashMap<u64, TransactionInputSigner>>,
+}
+
+impl Platform {
+    #[inline]
+    pub fn new(
+        coin: UtxoStandardCoin,
+        network: BlockchainNetwork,
+        default_fees_and_confirmations: PlatformCoinConfirmations,
+    ) -> Self {
+        Platform {
+            coin,
+            network,
+            best_block_height: AtomicU64::new(0),
+            default_fees_and_confirmations,
+            registered_txs: PaMutex::new(HashMap::new()),
+            registered_outputs: PaMutex::new(Vec::new()),
+            unsigned_funding_txs: PaMutex::new(HashMap::new()),
+        }
+    }
+
+    pub fn update_best_block_height(&self, new_height: u64) {
+        self.best_block_height.store(new_height, AtomicOrdering::Relaxed);
+    }
+
+    pub fn best_block_height(&self) -> u64 { self.best_block_height.load(AtomicOrdering::Relaxed) }
+
+    pub fn add_tx(&self, txid: Txid, script_pubkey: Script) {
+        let mut registered_txs = self.registered_txs.lock();
+        registered_txs
+            .entry(txid)
+            .or_insert_with(HashSet::new)
+            .insert(script_pubkey);
+    }
+
+    pub fn add_output(&self, output: WatchedOutput) {
+        let mut registered_outputs = self.registered_outputs.lock();
+        registered_outputs.push(output);
+    }
+
+    async fn get_tx_if_onchain(&self, txid: Txid) -> Result<Option<Transaction>, GetTxError> {
+        match self
+            .coin
+            .as_ref()
+            .rpc_client
```

Comment:
`rpc_client()` getter will help to make this shorter. `H256Json::from(txid.as_hash().into_inner()).reversed()` can also be done using a separate variable. I think it can become a one-liner afterward.